pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -725,29 +1058,37 @@ func shouldStartVM(instance *v2vv1.VirtualMachineImport) bool {
 	return instance.Spec.StartVM != nil && *instance.Spec.StartVM && conditions.HasSucceededConditionOfReason(instance.Status.Conditions, v2vv1.VirtualMachineReady)
 }
 
-// manageDataVolumeState update current state according to progress of import
-func (r *ReconcileVirtualMachineImport) manageDataVolumeState(instance *v2vv1.VirtualMachineImport, dvsDone map[string]bool, numberOfDvs int) (bool, error) {
+func shouldConvertGuest(instance *v2vv1.VirtualMachineImport) bool {
```

Comment:
I'd prefer to encapsulate this Provider - since this is provider specific and the controller after the point in which it creates the providers doesn't evaluate the Spec.Source for the provider type.